app/src/main/java/com/example/farmeraid/home/add_edit_produce/views/AddEditProduceScreenView.kt

```diff
@@ -0,0 +1,114 @@
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.filled.ArrowBack
+import androidx.compose.material3.CenterAlignedTopAppBar
+import androidx.compose.material3.ExperimentalMaterial3Api
+import androidx.compose.material3.Icon
+import androidx.compose.material3.IconButton
+import androidx.compose.material3.Scaffold
+import androidx.compose.material3.Text
+import androidx.compose.material3.TextField
+import androidx.compose.material3.TextFieldDefaults
+import androidx.compose.material3.TopAppBarDefaults
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.collectAsState
+import androidx.compose.runtime.getValue
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.text.input.KeyboardType
+import androidx.compose.ui.text.style.TextOverflow
+import androidx.compose.ui.unit.dp
+import androidx.hilt.navigation.compose.hiltViewModel
+import com.example.farmeraid.home.add_edit_produce.AddEditProduceViewModel
+import com.example.farmeraid.ui.theme.PrimaryColour
+import com.example.farmeraid.ui.theme.WhiteContentColour
+import com.example.farmeraid.uicomponents.ButtonView
+import com.example.farmeraid.uicomponents.models.UiComponentModel
+
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun AddEditProduceScreenView() {
+    val viewModel = hiltViewModel<AddEditProduceViewModel>()
+    val state by viewModel.state.collectAsState()
+
+    Scaffold(
+        topBar = {
+            CenterAlignedTopAppBar(
+                title = {
+                    if (state.isAddProduce) {
+                        Text(
+                            text = "Add Produce",
+                            maxLines = 1,
+                            overflow = TextOverflow.Ellipsis
+                        )
+                    } else {
+                        Text(
+                            text = "Edit ${state.produceName}",
+                            maxLines = 1,
+                            overflow = TextOverflow.Ellipsis
+                        )
+                    }
+
+                },
+                navigationIcon = {
+                    IconButton(onClick = { viewModel.navigateBack() }){
+                        Icon(
+                            imageVector = Icons.Filled.ArrowBack,
+                            contentDescription = "",
+                            tint = WhiteContentColour,
+                        )
+                    }
+                },
+                colors = TopAppBarDefaults.smallTopAppBarColors(
+                    containerColor = PrimaryColour,
+                    titleContentColor = WhiteContentColour,
+                )
+            )
+        }
+    ) { paddingValues ->
+        Column(
+            modifier = Modifier.padding(paddingValues).padding(20.dp),
+            verticalArrangement = Arrangement.spacedBy(20.dp)
+        ) {
+            TextField(
+                label = { Text(text = "Produce Name") },
```

Comment:
@RishanR are we allowing the user to enter any random Produce Name, or should we have a dropdown to limit the type of produce? Similar to how it was in the Figma mockup

Reply:
Agreed I think even for the add/remove stuff in the speech recognizer we would want to know all the produces we are dealing with as it will let us figure out all the plurals. So it makes sense to do dropdown since we would know what produces we will have.

Reply:
But isn't add/edit produce where we actually define the produce? So how would we have a dropdown here

Reply:
Hmm, that's a good point. But maybe the way I understood it was, we are limiting the options of what produce/crops the farmer can harvest (which is not the best design, but it should be enough for our use case). These produce can be predetermined in the repository, maybe hard-coded or fetching a list of produce from some 3rd-party API.

If this is the case then we can try using a dropdown with a search bar (like how we did it on the quotas page).

Reply:
Ohhh I see, hmm not sure where we could get a huge produce list like that though. But we can look into it.

Reply:
In that case, should I hold off on merging this PR till we confirm that?

Reply:
I think for now u can go for the merging of this feature, but we can pull some sort of list of items and use that as the possible harvets a person can make. This way as sath mentioned it helps the speech recog and simplifies that implementation.